.github/pull_request_template.md

```diff
@@ -0,0 +1,43 @@
+#### What does the PR do?
+<!-- Describe your pull request here. Please read the text below the line, and make sure you follow the checklist.-->
+
+
+#### Checklist
+- [ ] Populated all fields in this template.
+- [ ] The PR title reflects the change.
+- [ ] Changes are described in the pull request, or an [existing issue is referenced](https://github.com/triton-inference-server/server/issues).
+- [ ] Added [test plan](#test-plan) and verified test passes.
+- [ ] Verified that the PR passes existing CI.
+- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+- [ ] CodeQL suggestions resolved
+- [ ] Lint Gods are pleased (i.e. pre-commit checks are green)
+- [ ] Optional: Additional screenshots for behavior/output changes with before/after.
+
+#### Type:
+- [ ] Feature
+- [ ] Bugfix
```

Comment:
Should we just use the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) list if that's what we're going for here?

```
[
  'build',
  'chore',
  'ci',
  'docs',
  'feat',
  'fix',
  'perf',
  'refactor',
  'revert',
  'style',
  'test'
];
```

https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional

Reply:
And do we want to enforce having it in the PR title for automatic release notes / change logs?

ex:
```
chore: Update PR Template
```

Reply:
like the idea of referencing an existing standard / practice

Reply:
It looks like there are some github actions that can do either/both:
1. enforce PR titles follow the structure `<conventional_commit_type>: <Title>`
2. enforce commit messages follow `<conventional_commit_type>: <commit_message>`

I think (2) may be overkill or more of a "shock", but (1) could be nice to start with for automating the check.